index.js

```diff
@@ -276,6 +276,143 @@ async function getCircuit(conv) {
   }
 }
 
+
+/**
+ * Set a logged on user's presence
+ */
+app.intent('set.presence', async conv => {
+  const circuit = await getCircuit(conv);
+  if (!circuit) {
+    return;
+  }
+  const { presenceType } = conv.contexts.input['setpresence_data'].parameters;
+  const presence = presenceType.toLowerCase();
+  
+  const { untilTime } = conv.contexts.input['setpresence_data'].parameters;
+  const { duration } = conv.contexts.input['setpresence_data'].parameters;
+
+  if (presence == 'available') {
+    await circuit.setPresenceAvailable()
+    conv.ask(`Your online presence is set to AVAILABLE. Anything Else?`);
+    conv.contexts.set('anything_else', 5);
+  } else if ((presence == 'dnd' || presence == 'do not disturb') && (untilTime != '' || duration != '')) { 
```

Comment:
Similarly use '!==' not '!=', change these everywhere. Our programming standards are typically to use ===

Reply:
presence === Circuit.Enums.PresenceState.DND since the sdk never returns a 'do not disturb' value remove that.

Reply:
!!untilTime.length && !!duration.length

Reply:
You can also do !untilTime || !duration